code/game/objects/structures/shower.dm

```diff
@@ -1,38 +1,61 @@
 #define SHOWER_FREEZING "freezing"
+#define SHOWER_FREEZING_TEMP 100
 #define SHOWER_NORMAL "normal"
+#define SHOWER_NORMAL_TEMP 300
 #define SHOWER_BOILING "boiling"
+#define SHOWER_BOILING_TEMP 400
+#define SHOWER_REACTION_MULTIPLIER 0.025
 
 /obj/machinery/shower
 	name = "shower"
-	desc = "The HS-451. Installed in the 2550s by the Nanotrasen Hygiene Division."
+	desc = "A shower for keeping clean. Passively replenishes itself with water when not in use."
 	icon = 'icons/obj/watercloset.dmi'
 	icon_state = "shower"
 	density = FALSE
 	use_power = NO_POWER_USE
+	//Is the shower on or off?
 	var/on = FALSE
+	//What temperature the shower reagents are set to.
 	var/current_temperature = SHOWER_NORMAL
+	//What sound will be played on loop when the shower is on and pouring water.
 	var/datum/looping_sound/showering/soundloop
+	//What reagent should the shower be filled with when initially built.
 	var/reagent_id = /datum/reagent/water
-	var/reaction_volume = 200
+	//How much reagent capacity should the shower begin with when built.
+	var/reagent_capacity = 200
+	///How many units the shower refills every second.
+	var/refill_rate = 0.5
+	/// Whether or not the shower's water reclaimer is operating.
+	var/can_refill = TRUE
+	/// Whether to allow players to toggle the water reclaimer.
+	var/can_toggle_refill = TRUE
 
 /obj/machinery/shower/Initialize()
 	. = ..()
-	create_reagents(reaction_volume)
-	reagents.add_reagent(reagent_id, reaction_volume)
-
+	create_reagents(reagent_capacity)
+	reagents.add_reagent(reagent_id, reagent_capacity)
 	soundloop = new(list(src), FALSE)
 
+	AddComponent(/datum/component/plumbing/simple_demand)
```

Comment:
I'm not a big fan of the plumbing integration, as discussed. It doesn't fit very well on shiptest. The other changes are fine; if they're isolated I think the PR will be fine.